File: 1-TwoSum/1-TwoSum.java
// Last updated: 01/09/2025, 01:07:02
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hmap = new HashMap<>();
        for(int i = 0 ; i < nums.length ; i++){
            if(hmap.containsKey(target - nums[i])){
                return new int[]{hmap.get(target - nums[i]), i};
            }
            hmap.put(nums[i], i);
        }
        return new int[]{};
    }
}

/*
0 -> 3
1 -> 5
2 -> 9
*/

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 01/09/2025, 01:07:03
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>unmap;
        for(int i = 0 ; i < nums.size() ; i++){
            if(unmap.find(target - nums[i]) != unmap.end()){
                return {i , unmap[target - nums[i]]};
            }
            unmap[nums[i]] = i;
        }
        return {};
    }
};